api/src/main/scala/pl/touk/nussknacker/engine/api/runtimecontext/EngineRuntimeContext.scala

```diff
@@ -0,0 +1,9 @@
+package pl.touk.nussknacker.engine.api.runtimecontext
+
+import pl.touk.nussknacker.engine.util.metrics.MetricsProvider
+
+trait EngineRuntimeContext {
```

Comment:
Currently `JobData` are passed next to runtime context but are the part of runtime context. Take a look how the bests like Flink does - you have `RuntimeContext.getJobId`, `RuntimeContext.getTaskName`. Currently you will enforce consumers of our api to pass them both together...

Reply:
done